src/components/scenes/CoinRankingDetailsScene.tsx

```diff
@@ -218,6 +271,176 @@ const CoinRankingDetailsSceneComponent = (props: Props) => {
     return rows
   }
 
+  const getWallet = async () => {
+    if (edgeAsset == null) return
+    const { pluginId, tokenId } = edgeAsset
+
+    // Find all wallets that can hold this asset
+    const matchingWallets = Object.values(account.currencyWallets).filter(wallet => wallet.currencyInfo.pluginId === pluginId)
+    if (matchingWallets.length === 0) return
+
+    // If multiple wallets, show picker
+    let targetWallet = matchingWallets[0]
+    if (matchingWallets.length > 1) {
+      const result = await Airship.show<WalletListResult>(bridge => (
+        <WalletListModal
+          bridge={bridge}
+          navigation={navigation as NavigationBase}
+          headerTitle={lstrings.select_wallet_to_send_from}
+          allowedAssets={[{ pluginId, tokenId }]}
+          showCreateWallet
+        />
+      ))
+      if (result?.type !== 'wallet') return
+      targetWallet = account.currencyWallets[result.walletId]
+    }
+
+    // If no wallet exists, create one
+    if (targetWallet == null) {
+      const walletName = getUniqueWalletName(account, pluginId)
+      targetWallet = await createWallet(account, {
+        name: walletName,
+        walletType: `wallet:${pluginId}`,
+        fiatCurrencyCode: fiatCurrencyCode
+      })
+    }
+
+    return targetWallet
+  }
+
+  const handleBuyPress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const wallet = await getWallet()
+    if (wallet == null) return
```

Comment:
If the user presses the buy button, but the wallet is missing, nothing will happen. However, it seems like the function will only return `undefined` if the user cancels?

In any case, the `getWallet` function needs a documentation comment explaining the return value.

Reply:
The wallet will only be missing if the user pressed the button, was presented
with the wallet picker, and then dismissed the wallet picker without choosing anything